container-host-files/opt/scf/bin/klog.sh

```diff
@@ -22,152 +22,40 @@ if [ "$1" == "-f" ]; then
-    if grep -c -r -F "${PATTERN}" > .grep; then
-        [ "${NEWLINE}" == "1" ] && echo
-        NEWLINE=1
-
-        echo ">>> ${PATTERN}"
-        echo
-        grep -v :0$ .grep
-
-        while read INFO; do
-            echo ${INFO}
-        done
-    fi
-}
-
-# any pod with consul agent /var/vcap/sys/log/consul_agent/consul_agent.stdout.log
-lookfor <<EOF
-[ERR] agent: failed to sync remote state: No known Consul servers
-
-This happened on a consul agent that came up before the consul server,
-then abandoned the server as non-functional and then had no servers to
-talk with, restarting the pod fixed the issue (though restarting consul
-would also fix it).
-EOF
-
-read -d '' MSG <<EOF || true
-SST has happened twice and should only ever happen once. Probably missing the
-IST patch. If that patch is in place then full recovery of MySQL required:
-
-monit stop all # Check for lingering processes and kill them
-rm -rf /var/vcap/store/mysql/
-/var/vcap/jobs/mysql/bin/pre-start
-monit start all
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[ERROR] WSREP: SST failed: 1 (Operation not permitted)
-
-${MSG}
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] InnoDB: Cannot open table mysql/gtid_slave_pos from the internal data dictionary of InnoDB
-
-${MSG}
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] WSREP: no nodes coming from prim view, prim not possible
-
-Nodes all shut down improperly, MySQL must be manually rebootstrapped. Will fail
-if the IST patch has not been applied.
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] WSREP: Failed to prepare for incremental state transfer
-
-IST addresses are misconfigured, likely a <= 4.0.1 cluster, manual
-patch must be applied and MySQL must be recovered manually.
-
-Not sure exactly when/why this occurs. Perhaps when the primary node
-has gone away (as was the case here). The only restitution seems to be
-to full recover the mysql node (and ensure the IST patch is done):
-
-monit stop all # Check for lingering processes and kill them
-rm -rf /var/vcap/store/mysql/
-/var/vcap/jobs/mysql/bin/pre-start
-monit start all
-EOF
-
-read -d '' MSG <<EOF || true
-These point to an etcd that came up in a weird way and thinks there
-are other nodes when there are none. Restoration was done by:
-monit stop all
-rm -rf /var/vcap/store/etcd
-# Restart the pod using kubectl
-EOF
-
-# etcd pod /var/vcap/sys/log/etcd/etcd_ctl.err.log
-lookfor <<EOF
-Error:  cannot sync with the cluster using endpoints https://etcd-
-
-${MSG}
-EOF
+if [ ! -f "${DONE}" ]; then
+    mkdir -p "${KLOG}/${NS}"
+    rm -rf "${KLOG:?}/${NS:?}/"*
 
-# etcd pod /var/vcap/sys/log/etcd/etcd.stderr.log
-# duplicate of previous error?
-lookfor <<EOF
-etcdserver: publish error: etcdserver: request timed out
+    PODS=$(kubectl get pods --namespace "${NS}" --output name --show-all=true | sed 's/pods\///')
```

Comment:
If we make this an array `PODS=($(kubectl …))`, we can use `"${PODS[@]}"` later for safer quoting. Not that it actually makes a difference…

Reply:
:+1: 

Fixed in 6df1b836